docs/architecture/MLOps_process.md

```diff
@@ -0,0 +1,207 @@
+
+That’s all about the pipelines. Now, let’s discuss how to implement the development process for ML pipelines.
+
+## Development Process ##
+
+From a DevOps perspective the development process can be divided into three different stages:
+
+* Publish and Execute a pipeline from a local computer (or VM) using the full dataset or just a subset of data. Usually, it’s happening from a local feature branch where a developer or a data scientist is tuning the pipeline. Note that we are executing the pipeline _from_ a local computer rather than _on_ a local computer.
+* Create and validate Pull Request for changes from a feature branch towards to the development branch.
+* Publish and execute the pipeline in the development environment based on the full dataset.
+
+From implementation perspective our pipeline can be represented by a single Python file (for example, create_training_pipeline.py) that references all needed code and creates steps. You can build this file in a way that can be reused in the above three stages. We suggest the following:
+
+* Host all different parameters that are required for pipeline publishing as environment variables. This way you can initialize your local environment and publish the pipeline from a local computer or you can use a variable group in your favorite DevOps system to publish the pipeline.
+* Invoke the pipeline publishing script as a module. This way you can import any code for pipeline steps without any workarounds.
+* Use .amlignore to reduce number of files that you are submitting to Azure ML Service with your pipeline run. This file has the same syntax as .gitignore.
+
+### Branch name utilization ###
+
+The biggest challenge in ML pipeline development is how to modify and test the same pipeline from different branches. If we use fixed names for all experiments, models and pipeline names, it will be hard to differentiate the artifacts when working in a large team. To make sure that we can locate all feature branch related experiments, use feature branch name to mark all pipelines, experiment runs, and related artifacts. This way we can differentiate pipelines from different branches and data scientists can log various feature branch runs under the same name.
+
+The following example shows how to define Python variables based on initial environment variables for a branch name and pipeline type. Pipeline type is useful when you have multiple pipelines:
+
+```py
+pipeline_type = os.environ.get('PIPELINE_TYPE')
+source_branch = os.environ.get('BUILD_SOURCEBRANCHNAME')
+
+model_name = f"{pipeline_type}_{os.environ.get('MODEL_BASE_NAME')}_{source_branch}"
+pipeline_name = f"{pipeline_type}_{os.environ.get('PIPELINE_BASE_NAME')}_{source_branch}"
+experiment_name = f"{pipeline_type}_{os.environ.get('EXPERIMENT_BASE_NAME')}_{source_branch}"
+
+if len(experiment_name) > 36:
+    experiment_name = experiment_name[0:36]
```

Comment:
I would add a `logging.warning` here to notify the caller that the names have been truncated (and will possibly collide)

Reply:
printed out the names.